devsetup/scripts/tripleo.sh

```diff
@@ -56,24 +87,46 @@ sudo dnf install -y https://kojihub.stream.centos.org/kojifiles/packages/podman/
 sudo hostnamectl set-hostname undercloud.localdomain
 sudo hostnamectl set-hostname undercloud.localdomain --transient
 
+cat >\$HOME/nova_noceph.yaml <<__EOF__
+parameter_defaults:
+    NovaEnableRbdBackend: false
+__EOF__
+
 export HOST_PRIMARY_RESOLV_CONF_ENTRY=${HOST_PRIMARY_RESOLV_CONF_ENTRY}
 export INTERFACE_MTU=${INTERFACE_MTU:-1500}
 export NTP_SERVER=${NTP_SERVER:-"clock.corp.redhat.com"}
 export IP=${IP}
+export IP_ADRESS_SUFFIX=${IP_ADRESS_SUFFIX}
 export GATEWAY=${GATEWAY}
-export EDPM_COMPUTE_CEPH_ENABLED=${EDPM_COMPUTE_CEPH_ENABLED:-false}
-export CEPH_ARGS="${CEPH_ARGS:--e \$HOME/deployed_ceph.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/cephadm/cephadm.yaml}"
+export EDPM_COMPUTE_CEPH_ENABLED=${COMPUTE_CEPH_ENABLED:-False}
+export EDPM_COMPUTE_CEPH_NOVA=${COMPUTE_CEPH_NOVA:-False}
+export CEPH_ARGS="${CEPH_ARGS:--e \$HOME/deployed_ceph.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/cephadm/cephadm-rbd-only.yaml}"
+[[ "\$EDPM_COMPUTE_CEPH_NOVA" =~ "alse" ]] && export CEPH_ARGS="\${CEPH_ARGS} -e \$HOME/nova_noceph.yaml"
+export EDPM_COMPUTE_CELLS=${COMPUTE_CELLS:-1}
 
+set +x
 if [[ -f \$HOME/containers-prepare-parameters.yaml ]]; then
     echo "Using existing containers-prepare-parameters.yaml - contents:"
     cat \$HOME/containers-prepare-parameters.yaml
 else
+    login_args=" "
+    [ "\$RH_REGISTRY_USER" ] && [ -n "\$RH_REGISTRY_PWD" ] && login_args="--enable-registry-login"
     openstack tripleo container image prepare default \
-        --output-env-file \$HOME/containers-prepare-parameters.yaml
+        --output-env-file \$HOME/containers-prepare-parameters.yaml \${login_args}
     # Use wallaby el9 container images
     sed -i 's|quay.io/tripleowallaby$|quay.io/tripleowallabycentos9|' \$HOME/containers-prepare-parameters.yaml
 fi
 
+if [ "\$RH_REGISTRY_USER" ] && [ -n "\$RH_REGISTRY_PWD" ]; then
+    grep -q ContainerImageRegistryCredentials /home/zuul/containers-prepare-parameters.yaml || \
+    cat >> /home/zuul/containers-prepare-parameters.yaml <<__EOF__
+  ContainerImageRegistryCredentials:
+    registry.redhat.io:
+      \${RH_REGISTRY_USER}: \$RH_REGISTRY_PWD
```

Comment:
same concern here about leaking the registry user/pass

Reply:
> even with +x i don't think this is acceptable since you are actually writing the user/pass to file here
> this script is being executed by the existing multinode ci jobs i.e. not just for your new multi cell addition.
> Isn't the whole point of this work that we will be adding a new multi cell job in ci? So what do you mean about  local setups.

As a developer, I need a way to deploy it locally, using this same code.
It won't fly without registry creds. There is no reproducer for it in CI, or any docs on how to use it without auto gold nodes in zuul (not the best choice neither)

Reply:
We can make a patch to ci_framework to always unset ``RH_REGISTRY_PWD`` as adependency of this change. Then it should address your concen

Reply:
Done! Does it make sense now https://review.rdoproject.org/r/c/rdo-jobs/+/53192/34..35/playbooks/data_plane_adoption/install_yamls_deploy_tripleo.yaml ?

Reply:
PTAL @marios

Reply:
ack thanks @bogdando. I think https://review.rdoproject.org/r/c/rdo-jobs/+/53192/34..35/playbooks/data_plane_adoption/install_yamls_deploy_tripleo.yaml should make sure we aren't setting this in the ci runs.

It will be confusing though because someone will rightly ask "why are you setting the username and password of the registry to '' " ?

Thanks for acknowledging the concern and sanity checking that we aren't going to run this in ci.